### PR TITLE
chore(jenkins,circleci): Refactor APIs to export modules from index to support ADR004

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Refactors the API endpoints for the Jenkins and CircleCI plugins to use the result of ADR 004, exporting `*` from their `/api` folder packages.

The Jenkins plugin works on my local instance, though I've not directly tested CircleCI but test suites pass.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by adamdmharvey** [comment]
Reviewing TSC issues....

File: plugins/circleci/src/api/index.ts [hunk 1]
```diff
@@ -14,82 +14,4 @@
...
-      ...options.vcs,
-    });
-  }
-
-  private async getApiUrl() {
-    const proxyUrl = await this.discoveryApi.getBaseUrl('proxy');
-    return proxyUrl + this.proxyPath;
-  }
-}
+export * from './CircleCIApi';
```
Comment (Fox32) [inline plugins/circleci/src/api/index.ts:17]: The ADR says:

> Index files that are re-exporting symbols from non-index files should always enumerate all exports, that is:

So I would expect 

```typescript
export {CircleCIApi, ...} from './CircleCIApi';
```

here

File: plugins/jenkins/src/api/index.ts [hunk 2]
```diff
@@ -14,220 +14,4 @@
...
-      trimmedBuild.length - buildNumber.toString(10).length - 1,
-    );
-
-    return {
-      jobName,
-      buildNumber,
-    };
-  }
-}
+export { JenkinsApi, jenkinsApiRef } from './JenkinsApi';
```
Comment (Fox32) [inline plugins/jenkins/src/api/index.ts:17]: Yeah, like here 👍

**Review by Fox32 — COMMENTED** [review]
I'm not 100% sure if ADR004 states that index files are only allowed to have export, but no definitions. I think not, which would mean the as-is state is sufficient. Introducing an index file would make sense if the other file is a `tsx` file, or if there are more than one file in the same directory.

File: plugins/circleci/src/api/index.ts (hunk 1 above)
Comment (adamdmharvey) [reply plugins/circleci/src/api/index.ts]: Yah makes sense!  I think the doco threw me off a bit. As I go through the code, I've definitely found the "walking up the path" useful, and when I realized in Jenkins it wasn't that way for the API (or in CircleCI), just thought I'd give a shot to better understand it. Thanks for your patience, @Fox32!  Updates added.

**Review by freben — APPROVED** [review]
nice!

**Pull request merged**